lib/galaxy/files/sources/__init__.py

```diff
@@ -94,13 +94,13 @@ class FilesSourceOptions:
     # Indicates access to the FS operation with intent to write.
     # Even if a file source is "writeable" some directories (or elements) may be restricted or read-only
     # so those should be skipped while browsing with writeable=True.
-    writeable: Optional[bool]
+    writeable: Optional[bool] = False
 
     # Property overrides for values initially configured through the constructor. For example
     # the HTTPFilesSource passes in additional http_headers through these properties, which
     # are merged with constructor defined http_headers. The interpretation of these properties
     # are filesystem specific.
-    extra_props: Optional[FilesSourceProperties]
+    extra_props: Optional[FilesSourceProperties] = {}
```

Comment:
I think this is fine if you make the class a dataclass ... which may have been the intention from the start ?
Otherwise it's error prone to have this be a class var.

Reply:
Good point. fixed.